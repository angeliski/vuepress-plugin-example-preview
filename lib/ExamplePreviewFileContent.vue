<template>
  <div v-html="content" class="code" :class="'language-' + language"></div>
</template>

<script>
import highlightCode from '@vuepress/markdown/lib/highlight.js'

export default {
  props: {
    file: {
      type: Object,
      required: true,
    },
  },

  computed: {
    language: ({ file: { name }}) => name.substring(name.lastIndexOf('.') + 1, name.length),
    content: ({ file, language}) => highlightCode(file.content, language)
  }
}
</script>

<style scoped>
.code {
  border-radius: 0 0 4px 4px;
}

.code pre {
  margin: 0;
}

@media (max-width: 419px) {
  .code.code.code {
    margin: 0;
  }
}
</style>
