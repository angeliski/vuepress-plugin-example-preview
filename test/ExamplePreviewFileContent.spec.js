import ExamplePreviewFileContent from '../lib/ExamplePreviewFileContent'
import { mount } from '@vue/test-utils'

describe('ExamplePreviewFileContent ', () => {
  it('works', () => {
    const wrapper = mount(ExamplePreviewFileContent, {
      propsData: {
        file: {
          name: 'index.js',
          content: 'let a = 0',
        },
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
