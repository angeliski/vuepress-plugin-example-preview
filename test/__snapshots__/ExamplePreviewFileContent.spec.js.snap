// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExamplePreviewFileContent  works 1`] = `"<div class=\\"code language-js\\"><pre v-pre=\\"\\" class=\\"language-js\\"><code><span class=\\"token keyword\\">let</span> a <span class=\\"token operator\\">=</span> <span class=\\"token number\\">0</span></code></pre></div>"`;
